src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,85 +16,66 @@ public static class NUnitTestExtractorApp
-                            Directory.CreateDirectory(directory);
+                            fi.Directory.Create();
                         }
 
-                        if (!File.Exists(_options.Output))
+                        // if the output file doesn't exist, then create it
+                        if (!fi.Exists)
                         {
-                            using (FileStream stream = File.Create(_options.Output)) { }
+                            fi.Create();
                         }
                     }
 
                     if (_options.Level != null)
                     {
-                        StreamWriter writer = null;
-                        level = ParseLevel(_options.Level);
-                        try
-                        {
-                            if (_options.Output == null)
-                            {
-                                writer = new StreamWriter(Console.OpenStandardOutput());
-                            }
-                            else
-                            {
-                                writer = new StreamWriter(_options.Output, append: true);
-                            }
-
-                            GetTests(_options.DLLs, level, writer);
-                        }
-                        finally
-                        {
-                            if (writer != null)
-                            {
-                                writer.Dispose();
-                                
-                            }
-                        }
+                        Level level = ParseLevel(_options.Level);
+
+                        GetTests(_options.DLLs, level);
                     }
                 }
             }
         }
 
         /// <summary>
-        /// Writes each DLL file path along with either a namespace, class or function which contains tests(dependent on level specified by user)
+        /// Parses string into proper Level
         /// </summary>
-        /// <param name="writer"> StreamWriter used to write the dll path and data, either to a txt file or stdout </param>
-        /// <param name="dll"> The dll path which is written </param>
-        /// <param name="data"> The test information corrosponding to the dll file </param>
-        private static void WriteTestDllAndName(TextWriter writer, string dll, string data)
+        /// <param name="levelToParse">The string to parse</param>
+        /// <returns>The Level from parsing the string</returns>
+        public static Level ParseLevel(string levelToParse)
         {
-             writer.WriteLine("\""+dll+"\"" + " | " + data);
-             
+            try
+            {
+                return (Level)Enum.Parse(typeof(Level), levelToParse, ignoreCase: true);
+            }
+            catch
+            {
+                string errorMessage = string.Format("Incorrect argument - Level: {0}", levelToParse);
+
+                throw new Exception(errorMessage);
```

Comment:
[Could] Maybe throw something more specific like ArgumentException or ArgumentOutOfRangeException

Reply:
These kinds of base exception type have been changed in the next version.